libs/orbit-essentials/src/backup_snapshots.rs

```diff
@@ -0,0 +1,39 @@
+use orbit_essentials_macros::storable;
+use std::collections::VecDeque;
+
+#[storable]
+#[derive(Clone, Debug, Eq, PartialEq)]
+pub struct BackupSnapshots {
+    max_backup_snapshots: usize,
+    backup_snapshots: VecDeque<Vec<u8>>,
+}
+
+impl Default for BackupSnapshots {
+    fn default() -> Self {
+        Self::new(1)
+    }
+}
+
+impl BackupSnapshots {
+    pub fn new(max_backup_snapshots: usize) -> Self {
+        Self {
+            max_backup_snapshots,
+            backup_snapshots: VecDeque::new(),
+        }
+    }
+
+    pub fn replace_snapshot(&self) -> Option<Vec<u8>> {
```

Comment:
I think since "replace" is a verb, it suggests this function will replace a snapshot. I think a more suitable name would be "get_snapshot_to_replace". Same as regarding `SystemInfo::replace_upgrader_backup_snapshot`.